### PR TITLE
Support http.enabled in beats

### PR DESCRIPTION
**Pull request by v1v**
## What does this PR do?

When working on fixing the healthcheck, @graphaelli suggested to enable [`http.enabled`](https://www.elastic.co/guide/en/beats/filebeat/master/http-endpoint.html)

## Why is it important?

Healthcheck the services before using them.

## Issues

Caused by https://github.com/elastic/apm-integration-testing/pull/990

Closes https://github.com/elastic/apm-integration-testing/issues/986

## Tests

```bash
$. ./scripts/compose.py start master \
         --with-filebeat \
         --with-metricbeat \
         --with-heartbeat \
         --force-build
```

Then all the services are healthy:

```bash
$ docker ps
CONTAINER ID        IMAGE                                                          COMMAND                  CREATED              STATUS                        PORTS                                                                                                        NAMES
f8a057d3fe45        docker.elastic.co/apm/apm-server:8.0.0-SNAPSHOT                "/usr/local/bin/dock…"   25 seconds ago       Up 22 seconds (healthy)       127.0.0.1:6060->6060/tcp, 127.0.0.1:8200->8200/tcp, 127.0.0.1:14250->14250/tcp, 127.0.0.1:14268->14268/tcp   localtesting_8.0.0_apm-server
a4a2a2dc71eb        docker.elastic.co/beats/metricbeat:8.0.0-SNAPSHOT              "/usr/local/bin/dock…"   25 seconds ago       Up 22 seconds (healthy)                                                                                                                    localtesting_8.0.0_metricbeat
5e31cf73b433        docker.elastic.co/beats/filebeat:8.0.0-SNAPSHOT                "/usr/local/bin/dock…"   25 seconds ago       Up 23 seconds (healthy)                                                                                                                    localtesting_8.0.0_filebeat
27a2ae79933e        docker.elastic.co/beats/heartbeat:8.0.0-SNAPSHOT               "/usr/local/bin/dock…"   46 seconds ago       Up 41 seconds (healthy)                                                                                                                    localtesting_8.0.0_heartbeat
6c372d617da0        docker.elastic.co/kibana/kibana:8.0.0-SNAPSHOT                 "/bin/tini -- /usr/l…"   46 seconds ago       Up 45 seconds (healthy)       127.0.0.1:5601->5601/tcp                                                                                     localtesting_8.0.0_kibana
ab6ac5dbe331        docker.elastic.co/elasticsearch/elasticsearch:8.0.0-SNAPSHOT   "/bin/tini -- /usr/l…"   About a minute ago   Up About a minute (healthy)   127.0.0.1:9200->9200/tcp, 9300/tcp                                                                           localtesting_8.0.0_elasticsearch

```

```json
$ docker inspect  localtesting_8.0.0_filebeat | jq '.[].State.Health'
{
  "Status": "healthy",
  "FailingStreak": 0,
  "Log": [
    {
      "Start": "2020-11-20T10:20:56.914173759Z",
      "End": "2020-11-20T10:20:56.994095382Z",
      "ExitCode": 0,
      "Output": "'HTTP 200'"
    },
    {
      "Start": "2020-11-20T10:21:06.999721994Z",
      "End": "2020-11-20T10:21:07.087476887Z",
      "ExitCode": 0,
      "Output": "'HTTP 200'"
    },
    {
      "Start": "2020-11-20T10:21:17.092060463Z",
      "End": "2020-11-20T10:21:17.194326011Z",
      "ExitCode": 0,
      "Output": "'HTTP 200'"
    }
  ]
}
```

### Test with a bunch more of flags

```bash
$ ./scripts/compose.py start \
    --release \
    --with-opbeans-dotnet \
    --with-opbeans-go \
    --with-opbeans-java \
    --opbeans-java-agent-branch=pr/588/head \
    --force-build \
    --with-opbeans-node \
    --with-opbeans-python \
    --with-opbeans-ruby \
    --with-opbeans-rum \
    --with-filebeat \
    --with-metricbeat \
    --with-packetbeat \
    --with-heartbeat \
    7.10.0
```

## Tasks

- [x] Test packetbeat. See https://github.com/elastic/apm-integration-testing/pull/993
- [ ] Do we wanna support `.*.6.x-compat` ?

### CODE REVIEW

**Comment by kuisathaverat** [comment]
Where did you saw this magic spell?  😄

**Review by kuisathaverat — APPROVED** [review]

File: docker/heartbeat/heartbeat.yml [hunk 1]
```diff
@@ -44,3 +44,7 @@ heartbeat.autodiscover:
               urls: ["http://${data.host}:${data.port}"]
               schedule: "@every 5s"
               timeout: 1s
+
+http.enabled: true
+http.host: localhost
+http.port: 5066
```
Comment (cachedout) [inline docker/heartbeat/heartbeat.yml:50]:
```suggestion
http.port: 5066

```
Super-nit: newline consistency :)

**Review by cachedout — APPROVED** [review]
👍 I really like this idea.

**Review by graphaelli — APPROVED** [review]

**Comment by v1v** [comment]
I'll merge this PR since the issue is somehow related with the maven central when building the java agent from source code :/

**Pull request merged**